docs/docs/guides/ecosystem/langchain.md

```diff
@@ -0,0 +1,122 @@
+
+
+## Getting Started
+
+To get started, simply call `weave.init()` at the beginning of your script. The argument in weave.init() is a project name that will help you organize your traces.
+
+```python
+import weave
+from langchain_core.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+
+# Initialize Weave with your project name
+# highlight-next-line
+weave.init("langchain_demo")
+
+llm = ChatOpenAI()
+prompt = PromptTemplate.from_template("1 + {number} = ")
+
+llm_chain = prompt | llm
+
+output = llm_chain.invoke({"number": 2})
+
+print(output)
+```
+
+## Traces
+
+Storing traces of LLM applications in a central database is crucial during both development and production. These traces are essential for debugging and improving your application by providing a valuable dataset.
+
+Weave automatically captures traces for your LangChain applications. It will track and log all calls made through the LangChain library, including prompt templates, chains, LLM calls, tools, and agent steps. You can view the traces in the Weave web interface.
+
+[![langchain_trace.png](imgs/langchain_trace.png)](https://wandb.ai/parambharat/langchain_demo/weave/calls)
+
+## Manually Tracing Calls
+
+In addition to automatic tracing, you can manually trace calls using the `WeaveTracer` callback or the `weave_tracing_enabled` context manager. These methods are akin to using request callbacks in individual parts of a LangChain application.
+
+### Using `WeaveTracer`
+
+You can pass the `WeaveTracer` callback to individual LangChain components to trace specific requests.
+
+```python
+from weave.integrations.langchain import WeaveTracer
+from langchain_core.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+import weave
+
+# Initialize Weave with your project name
+# highlight-next-line
+weave.init("langchain_demo")
+
+weave_tracer = WeaveTracer()
+
+config = {"callbacks": [weave_tracer]}
+
+llm = ChatOpenAI()
+prompt = PromptTemplate.from_template("1 + {number} = ")
+
+llm_chain = prompt | llm
+
+output = llm_chain.invoke({"number": 2}, config=config)
+
+print(output)
+```
+
+### Using `weave_tracing_enabled` Context Manager
+
+Alternatively, you can use the `weave_tracing_enabled` context manager to enable tracing for specific blocks of code.
+
+```python
+from weave.integrations.langchain import weave_tracing_enabled
+from langchain_core.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+import weave
+
+# Initialize Weave with your project name
+# highlight-next-line
+weave.init("langchain_demo")
+
+llm = ChatOpenAI()
+prompt = PromptTemplate.from_template("1 + {number} = ")
+
+llm_chain = prompt | llm
+
+with weave_tracing_enabled():
+    output = llm_chain.invoke({"number": 2})
+
+print(output)
+```
```

Comment:
@parambharat these are great! Could you also introduce the concept of a Model here? Considering they may be needed for Evaluation, it'd be good to have an example that shows a way for users to create instrument their code so that they get a nice evaluation table.